[Bug]: DebeziumIO fails to claim records from a database

### What happened?

When I ran DebeziumIO.read() (2.45.0 and higher) transformation with default configuration it fails with exception "Error occurred when consuming changes from Database". In fact it throws NullPointerException when trying to unbox "restriction.milisToRun" Integer (which is null) when trying to claim records from a database. 

![image](https://github.com/apache/beam/assets/19977507/03a179c5-972e-4522-b0b2-8934c2089962)

This field is defined as nullable so each time it becomes null the NullPointerException is thrown here.

![image](https://github.com/apache/beam/assets/19977507/226ae535-e856-4193-ad97-40d22070c60f)

Stacktrace:
`Caused by: java.lang.RuntimeException: Error occurred when consuming changes from Database. 
	at org.apache.beam.io.debezium.KafkaSourceConsumerFn.process(KafkaSourceConsumerFn.java:269)
Caused by: java.lang.NullPointerException
	at org.apache.beam.io.debezium.KafkaSourceConsumerFn$OffsetTracker.tryClaim(KafkaSourceConsumerFn.java:403)
	at org.apache.beam.io.debezium.KafkaSourceConsumerFn$OffsetTracker.tryClaim(KafkaSourceConsumerFn.java:366)
	at org.apache.beam.repackaged.direct_java.sdk.fn.splittabledofn.RestrictionTrackers$RestrictionTrackerObserver.tryClaim(RestrictionTrackers.java:59)
	at org.apache.beam.io.debezium.KafkaSourceConsumerFn.process(KafkaSourceConsumerFn.java:254)
	at org.apache.beam.io.debezium.KafkaSourceConsumerFn$DoFnInvoker.invokeProcessElement(Unknown Source)
	at org.apache.beam.repackaged.direct_java.runners.core.OutputAndTimeBoundedSplittableProcessElementInvoker.invokeProcessElement(OutputAndTimeBoundedSplittableProcessElementInvoker.java:125)
	at org.apache.beam.repackaged.direct_java.runners.core.SplittableParDoViaKeyedWorkItems$ProcessFn.processElement(SplittableParDoViaKeyedWorkItems.java:567)
	at org.apache.beam.repackaged.direct_java.runners.core.SplittableParDoViaKeyedWorkItems$ProcessFn$DoFnInvoker.invokeProcessElement(Unknown Source)
	at org.apache.beam.repackaged.direct_java.runners.core.SimpleDoFnRunner.invokeProcessElement(SimpleDoFnRunner.java:211)
	at org.apache.beam.repackaged.direct_java.runners.core.SimpleDoFnRunner.processElement(SimpleDoFnRunner.java:188)
	at org.apache.beam.repackaged.direct_java.runners.core.ProcessFnRunner.processElementInReadyWindows(ProcessFnRunner.java:76)
	at org.apache.beam.runners.direct.ParDoEvaluator.processElement(ParDoEvaluator.java:244)
	at org.apache.beam.runners.direct.DoFnLifecycleManagerRemovingTransformEvaluator.processElement(DoFnLifecycleManagerRemovingTransformEvaluator.java:54)
	at org.apache.beam.runners.direct.DirectTransformExecutor.processElements(DirectTransformExecutor.java:165)
	at org.apache.beam.runners.direct.DirectTransformExecutor.run(DirectTransformExecutor.java:129)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)`



### Issue Priority

Priority: 2 (default / most bugs should be filed as P2)

### Issue Components

- [ ] Component: Python SDK
- [ ] Component: Java SDK
- [ ] Component: Go SDK
- [ ] Component: Typescript SDK
- [X] Component: IO connector
- [ ] Component: Beam examples
- [ ] Component: Beam playground
- [ ] Component: Beam katas
- [ ] Component: Website
- [ ] Component: Spark Runner
- [ ] Component: Flink Runner
- [ ] Component: Samza Runner
- [ ] Component: Twister2 Runner
- [ ] Component: Hazelcast Jet Runner
- [ ] Component: Google Cloud Dataflow Runner

--- a/sdks/java/io/debezium/src/main/java/org/apache/beam/io/debezium/KafkaSourceConsumerFn.java
+++ b/sdks/java/io/debezium/src/main/java/org/apache/beam/io/debezium/KafkaSourceConsumerFn.java
@@ -400,15 +400,13 @@ public class KafkaSourceConsumerFn<T> extends DoFn<Map<String, String>, T> {
       this.restriction.fetchedRecords = fetchedRecords;
       LOG.debug("-------------- History: {}", this.restriction.history);
 
-      if (this.restriction.maxRecords == null && this.restriction.milisToRun == -1) {
-        return true;
-      }
-
       // If we've reached the maximum number of records OR the maximum time, we reject
       // the attempt to claim.
       // If we've reached neither, then we continue approve the claim.
       return (this.restriction.maxRecords == null || fetchedRecords < this.restriction.maxRecords)
-          && (this.restriction.milisToRun == null || elapsedTime < this.restriction.milisToRun);
+          && (this.restriction.milisToRun == null
+              || this.restriction.milisToRun == -1
+              || elapsedTime < this.restriction.milisToRun);
     }
 
     @Override

